src/system/SystemPacketBufferInternal.h

```diff
@@ -82,7 +82,7 @@
  *
  * True if packet buffers are allocated from an LwIP custom pool.
  */
-#if CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_POOL && !LWIP_PBUF_FROM_CUSTOM_POOLS
+#if CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_POOL && !(defined(LWIP_PBUF_FROM_CUSTOM_POOLS) && LWIP_PBUF_FROM_CUSTOM_POOLS)
```

Comment:
Again, this is wrong.  The `#include <lwip/opt.h>` above should be pulling in those values.  This is breaking detection of cases when configs are not correctly set up.